enhancements/windows-containers/health-management.md

```diff
@@ -150,7 +159,8 @@ added to the cluster as a Node. The proposed schema is as follows:
       "name": "string within command field that will be substituted",
       "path": "location of the PowerShell script to be run, in order to get its output"
     },
-    "priority": "integer that will be used to order the creation of the services, priority 1 is created first"
+    "bootstrap": "boolean flag indicating whether this service should be handled as part of node bootstrapping",
+    "priority": "non-negative integer that will be used to order the creation of the services, priority 0 is created first"
```

Comment:
What if I have `priority=2, bootstrap=true` and `priority=1,boostrap=false`? Basically means you need to say more about the interplay of these two options.

Reply:
Good point. @sebsoto I think we need to add the restriction back, otherwise I'm not sure how we'd handle this case.
`Each service that has the bootstrap flag set as true must have a higher priority than all non-bootstrap services. WICD will throw an error if this is not the case.`

Reply:
Or you can sort them separately and add a restriction that only a controller service can depend on a bootstrap service but not the other way around.

Reply:
We will be creating `bootstrap` services before `controller` ones. So using the example above, if we have Service A `priority=2, bootstrap=true` and Service B `priority=1,boostrap=false`, we are basically saying the priority field can be meaningless when compared across WICD stages (since Service A will be started first, despite being lower priority).
This is fine imo but it's just not super clear. If I'm understanding it right, I can add an explanation to the enhancement?

Reply:
> the priority field can be meaningless

Thats a bit of an overstatement. The priority field allows us to find order in services. Bootstrapping is a special case, that will only happen once for a node, while the `controller` command, will be reconciling through the node's entire lifespan.

For example lets say you have 3 services with priorities {4,0,1}
Lets say the ordered bootstrap subset contains {0,4}. We use the priority field to order these services, just the same as we use it to order the ordered full set {0,1,4}.

Please go ahead and add an explanation

Reply:
Explanation added here: https://github.com/openshift/enhancements/pull/1018/files#diff-3c7c637b883f288bf20ea8e8e521c4962adef0c81b2d149955787c8baf87273fR184-R188

Reply:
>Lets say the ordered bootstrap subset contains {0,4}. We use the priority field to order these services, just the same as we use it to order the ordered full set {0,1,4}.

This will not work. By definition, `bootstrap` services have a higher priority than the `controller` ones.

Reply:
As discussed in standup:
```
Services marked with `bootstrap: true` will start before any others. Bootstrap services cannot depend on a non-bootstrap
service; WICD will throw an error if this is detected. Similarly, each service that has the bootstrap flag set as true
must have a higher priority than all non-bootstrap services. WICD will throw an error if this is not the case.
```